File: L1/c/kwadrat.c
#include <stdio.h>
#include <math.h>

int main(){
    float a = 0, b = 0, c = 0;

    printf("a = ");
    scanf("%f", &a);
    printf("b = ");
    scanf("%f", &b);
    printf("c = ");
    scanf("%f", &c);

    if(a == 0){
        if(b == 0){
            if(c == 0) printf("x należy do liczb rzeczywistych\n");
            else printf("x0 nie istnieje\n");
        }
        else printf("x0 = %f\n", -c / b);
    }
    else{
        float delta = b * b - 4 * a * c;

        if(delta > 0){
            printf("x1 = %f\n", (-b - sqrt(delta)) / 2 / a);
            printf("x2 = %f\n", (-b + sqrt(delta)) / 2 / a);
        }
        else if(delta == 0) printf("x0 = %f\n", -b / 2 / a);
        else{
            printf("Rozwiązanie nie jest w zbiorze liczb rzeczywistych\n");
        }
    }

    return 0;
}

File: L1/python/kwadrat.py
#!/usr/bin/env python
import math


def main():
    a = float(input("a = "))
    b = float(input("b = "))
    c = float(input("c = "))

    if a != 0:
        delta = b ** 2 - 4 * a * c
        if delta < 0:
            print("x nie jest liczbą rzeczywistą")
        elif delta == 0:
            x1 = -b / 2 / a
            print(f"x = {x1}")
        else:
            x1 = (-b - math.sqrt(delta)) / 2 / a
            x2 = (-b + math.sqrt(delta)) / 2 / a
            print(f"x1 = {x1}\nx2 = {x2}")

    elif b != 0:
        x1 = -c / b
        print(f"x = {x1}")
    elif c == 0:
        print("x należy do liczb rzeczywistych")
    else:
        printf("x nie istnieje")



if __name__ == "__main__":
    main()
